### ISSUE TITLE
Can't create a provisioner with an AWS launch template

### ISSUE THREAD

**Issue by woz5999** [aws/karpenter#1353]
## Version

**Karpenter**: v0.6.2

**Kubernetes**: v1.21.0

## Expected Behavior
I can create an provisioner using an aws launch template.

## Actual Behavior
Validation error:
```
Error: admission webhook "validation.webhook.provisioners.karpenter.sh" denied the request: validation failed: expected exactly one, got both: spec.provider.amiFamily, spec.provider.launchTemplate
```
The error message is right: if launchTemplate is specified, amiFamily shouldn't be. Accordingly, I am not setting amiFamily in the provisioner, so this seems to be a bug rather than a configuration issues. 

## Steps to Reproduce the Problem
Create a basic provisioner using an aws launch template with the latest version of karpenter

## Resource Specs and Logs
```
apiVersion: karpenter.sh/v1alpha5
kind: Provisioner
metadata:
  name: test-provisioner
spec:
  provider:
    launchTemplate: "some-launch-template"
    subnetSelector:
      Name: "somes-selector*"
```

**Comment by AntonYanchuk** [comment]
Met the same bug.

**Comment by oussemos** [comment]
Looks like it was fixed here https://github.com/aws/karpenter/pull/1352

**Comment by suket22** [comment]
We'll try and get another release out to fix this today

**Comment by nonoswz** [comment]
I am getting a similar issue, but I get one more clause in the error about instanceProfile/and security groups. So I am not sure if the fix would also fix this error. This is definitely blocking on our side. We would definitely appreciate the new release today if doable.

This is the error when trying to create a new provisioner from Terraform, so we are not able to create new provisioners: 
`│Error: admission webhook "validation.webhook.provisioners.karpenter.sh" denied the request: validation failed: expected exactly one, got both: spec.provider.amiFamily, spec.provider.instanceProfile, spec.provider.launchTemplate` 

Should the instanceProfile not be in the provisioner if the launch template is specified?

Old provisioners seems to be working, but can't seem to be reconciled, neither edited with a similar error.
`error: provisioners.karpenter.sh "default" could not be patched: admission webhook "validation.webhook.provisioners.karpenter.sh" denied the request: validation failed: expected exactly one, got both: spec.provider.amiFamily, spec.provider.instanceProfile, spec.provider.launchTemplate, spec.provider.securityGroupSelector`

This is the spec of the provisioner
````
spec:
  kubeletConfiguration: {}
  labels:
    group: default
  limits: {}
  provider:
    apiVersion: extensions.karpenter.sh/v1alpha1
    instanceProfile: <our_instance_profile>
    kind: AWS
    launchTemplate: <our_launch_template>
    securityGroupSelector:
      kubernetes.io/cluster/<cluster_name>: '*'
    subnetSelector:
      Name: <filterForSubnet>
  requirements:
  - key: karpenter.sh/capacity-type
    operator: In
    values:
    - on-demand
  - key: node.kubernetes.io/instance-type
    operator: In
    values:
    - m5.4xlarge
  - key: topology.kubernetes.io/zone
    operator: In
    values:
    - us-west-2a
    - us-west-2b
    - us-west-2c
  - key: kubernetes.io/arch
    operator: In
    values:
    - amd64
  ttlSecondsAfterEmpty: 30
  ttlSecondsUntilExpired: 86400
````

The securitySubnetSelector is added by default, we are not specifying any (we do in the launch template).

**Comment by woz5999** [comment]
Thanks for the quick turnaround!

**Comment by nonoswz** [comment]
@suket22 Thanks for the patch and the information. We were able to make it work. What happened was that the  `securityGroupSelector` was added at some point in the provisioner configuration, the webhook just patches the resource, meaning, it will leave the ` securityGroupSelector` when applying the new provisioner resource through Terraform. So with the validation mechanism, it was throwing the error. 

Removing manually the ` securityGroupSelector` from the provisioner spec did the trick.
